iqrm/core.py

```diff
@@ -93,6 +96,14 @@ def iqrm_mask(x, radius=5, threshold=3.0):
             votes_cast[j].add(i)
             votes_received[i].add(j)
 
+    # Ignore some channels
+    nchans = len(x)
+    for c in ignorechans:
+        other_chans = list(range(0,c)) + list(range(c+1, nchans))
+        votes_received[c] = other_chans
+        votes_cast[c] = other_chans
```

Comment:
Both these edits to the votes dictionaries are "asymmetrical". The correct way is:

```python
        votes_received[c] = other_chans
        for i in other_chans:
            votes_cast[i].add(c)

        votes_cast[c] = other_chans
        for i in other_chans:
            votes_received[i].add(c)
```

Reply:
As I said in my first message, just not touching the votes dictionaries and setting `mask[ignorechans] = True` could be much faster (and simpler!), but you'd have to confirm whether that works. I leave that decision to you, and it can be done in a future PR.